pkg/build/generator/generator.go

```diff
@@ -211,6 +211,9 @@ func (g *BuildGenerator) checkLastVersion(bc *buildapi.BuildConfig, lastVersion
 // updateImageTriggers sets the LastTriggeredImageID on all the ImageChangeTriggers on the BuildConfig and
 // updates the From reference of the strategy if the strategy uses an ImageStream or ImageStreamTag reference
 func (g *BuildGenerator) updateImageTriggers(ctx kapi.Context, bc *buildapi.BuildConfig, from, triggeredBy *kapi.ObjectReference) error {
+	if triggeredBy == nil {
+		return nil
```

Comment:
This should not be nil but an error right? If the updateImageTriggers failed because the triggeredBy is nil for some reason. We have to investigate why it is nil and give users clues why this function failed to update the triggers.